task.go

```diff
@@ -402,10 +415,14 @@ func (task *Task) Schedule() {
 	go func() {
 		select {
 		case <-time.After(fromNow):
-			task.queue.fire <- task
+			// It's possible the task might already be `done` if e.g. it was called by RunTask before the originally
+			// scheduled execution time. In that case this is a no-op, otherwise, run the task.
+			if !task.isDone {
```

Comment:
I don't quite get how this check avoids that problem - what if runtask is called at exactly this point?